src/components/newsletter/custom_button.js

```diff
@@ -0,0 +1,64 @@
+
+import {BUTTON_TYPE_VALUES,BUTTON_TYPE_DESCRIPTION}  from "../../modules/newsletters/button-module"
+import {useState} from 'react'
+
+import toast, { Toaster } from 'react-hot-toast';
+
+export default function CustomNewsletterButton(){
+// once subtype is selected here 
+    const [button_type,selectType] = useState(undefined)
+    const notify = (value) => toast(value);
+
+    function valueOrUndefined(value){
+        return value != undefined?value:undefined
+    }
+
+    const submitType=(e)=>{
+        e.preventDefault()
+        const new_values={
+            'button_type': valueOrUndefined(button_type),
+            'url_link': valueOrUndefined(e.target.button_link.value),
+            'button_text': valueOrUndefined(e.target.button_text.value)
+        }
+        console.log(new_values)
+        for(let [key,value] of Object.entries(new_values)){
+            if(value === undefined){
+                notify(key.split("_").join(" "), "is empty")
+            }
+        }
+        const final_text = '${"type":"'+new_values.button_type+'", "link":"'+new_values.url_link+'", "text":"'+new_values.button_text+'"}$'
+        navigator.clipboard.writeText(final_text)
```

Comment:
Byron wanted to see this piece of code :)

Reply:
I'm @vertex here!  Thank you!!